---
title: Trees and Beyond!
layout: presentation
reveal:
    theme: theme.css
---

<section class="titlepage center">
  <div class="title">
    Trees and Beyond!

    <div class="subtitle">
      Adventures in Bayesian Phylodynamics
    </div>

  </div>

  <div class="authors">
    Tim Vaughan
  </div>
  <div class="institution">Stadler Group, D-BSSE, ETH Z&uuml;rich</div>

  <div class="date">
    Biozentrum, University of Basel, 16<sup>th</sup> Sep, 2019
  </div>

  <div style="margin-top:1em">
    <div style="display:inline-block;">
      <img data-src="SIB_logo_20.svg" style="height:100px">
    </div>
    <div style="display:inline-block;">
      <img data-src="eth_logo.png" style="height:100px">
    </div>
    <div style="display:inline-block;">
      <img data-src="bsse_logo.gif" style="height:100px">
    </div>
    <div style="display:inline-block;">
      <img data-src="cEvo_logo.png" style="height:80px">
    </div>
  </div>
</section>

<section>
  <h1>The Computational Evolution (cEvo) Research Group</h1>

  <img data-src="cEvo_epi.jpg" style="float:left;width:48%">
  <img data-src="cEvo_macro.jpg" style="float:right;width:48%">
  <center>
    <img data-src="group_sushi_photo.jpg" style="width:50%">
  </center>
  
  
</section>

<section>
  <div class="flipper" style="width:100%;height:800px;margin-top:-2em">
  <img data-src="AdventureMap.png" style="height:100%;margin-top:0">
  <img data-src="AdventureMapStart.png" style="height:100%;margin-top:0" class="fragment">
  <img data-src="AdventureMapIntro.png" style="height:100%;margin-top:0" class="fragment">
  </div>
</section>

<section>
  <h1>What is a Population?</h1>

  <div style="align:center">

    <img style="height:250px;margin-right:20px;margin-bottom:0px" data-src="viruses.jpg">
    <img style="height:250px;margin-bottom:0px" data-src="lemurs.jpg">
  <img style="height:250px;margin-top:0px;padding-top:0px" data-src="species.jpg">
  </div>
</section>

<section>
  <h1>What is a Phylogenetic Tree?</h1>

  <div style="position:relative; width:800px; height:600px; margin:0 auto;">
    <div class="fragment current-visible"
         style="position:absolute">

      <br>

      <blockquote> A phylogenetic tree is a
        branching diagram or "tree" showing the evoluionary
        relationships between various species, individuals,
        or genes.
      </blockquote>
    </div>
    <img class="fragment current-visible" width="600" style="margin:0px;padding:0px;"
         data-src="primate_tree.jpg"
         style="position:absolute;top:0;left:0;" />
    <div class="figure fragment"
         style="position:absolute;top:0;left:0;margin:0;padding:0;">
      <img width="800" data-src="zika_tree.jpg" style="margin:0px;padding:0px"/>
      <div class="cite">Faria et al., Nature, 2017</div>
    </div>
  </div>
</section>

<section>
  <h1>What is Phylodynamics?</h1>
  <ul>
    <li>What does a phylogenetic tree tell us about a population?</li>
    <li>Places the emphasis on the population itself instead of the subset of individuals involved in a particular phylogeny.</li>
  </ul>

  <img data-src="phylodynamics.svg"/>
</section>

<section class="center">
  <h1>Connecting the dots...</h1>

  <div class="flipper">
    <img data-src="connecting_dots1.svg">
    <img data-src="connecting_dots2.svg" class="fragment">
    <img data-src="connecting_dots3.svg" class="fragment">
    <!--img data-src="connecting_dots4.svg" class="fragment"-->
    <img data-src="connecting_dots5.svg" class="fragment">
  </div>
</section>

<section>
  <h1>Bayesian phylodynamics</h1>

  <img data-src="Data_generation.svg" style="width:80%;text-align:center">

  <ul class="spaced">
    <li>Generative models connect the population dynamics model to the observed sequence alignment.</li>
    <li>Use Bayesian inference to work backwards:</li>
  </ul>
  <img data-src="Posterior_treeprior_nohl.svg">
</section>


<section>
  <div class="flipper" style="width:100%;height:800px;margin-top:-2em">
    <img data-src="AdventureMapEpiInf.png" style="height:100%;margin-top:0">
  </div>
</section>


<section>
  <h1>Modelling epidemics using birth-death processes</h1>


  <ul>
    <li>For small populations, <i>or populations which are temporarily small</i> (e.g. exponential growth), discrete nature of population size matters.</li>
    <li>As we usually have no way of knowing the exact timing of population events, this suggests use of stochastic population models.</li>
  </ul>

  <div class="fragment">
  <p>For example,</p>
  <blockquote class="textbox">
    <div class="title">The stochastic SIS model</div>
    \begin{align}
    S + I & \overset{\beta}{\longrightarrow} 2I\\
    I & \overset{\gamma}{\longrightarrow} S
    \end{align}
  </blockquote>
  </div>

  <ul>
    <li class="fragment">How do we compute $P(G|\beta,\gamma)$ for such models?</li>
    <li class="fragment">Can we jointly infer the population dynamics?</li>
  </ul>
</section>

<section>
  <h1>Particle Filtering for Trees</h1>

  <div class="flipper" style="height:500px">
    <img data-src="bigSIS.svg">
    <img data-src="treeBPF10.svg" class="fragment" data-fragment-index="1">
    <img data-src="treeBPF9.svg" class="fragment" data-fragment-index="2">
    <img data-src="treeBPF8.svg" class="fragment" data-fragment-index="3">
    <img data-src="treeBPF7.svg" class="fragment" data-fragment-index="4">
    <img data-src="treeBPF6.svg" class="fragment" data-fragment-index="5">
    <!--img data-src="treeBPF5.svg" class="fragment"-->
    <!--img data-src="treeBPF4.svg" class="fragment"-->
    <!--img data-src="treeBPF3.svg" class="fragment"-->
    <!--img data-src="treeBPF2.svg" class="fragment"-->
    <img data-src="treeBPF.svg" class="fragment" data-fragment-index="6">
  </div>

  <p class="fragment" data-fragment-index="6">Here $p_{\textrm{coal}}=\binom{k(t)}{2}/\binom{I(t)}{2}$ where $k(t)=$ num. of lineages just after $t$.</p>
  
</section>

<section>
  <h1>Particle Filtering for Trees</h1>

  <div class="flipper" style="height:800px">
    <img data-src="exampleTrajectories1.svg">
    <img data-src="exampleTrajectories2.svg" class="fragment">
  </div>
</section>

<section>
  <h1>Ebola in Sierra Leone</h1>
  <ul>
    <li>Aim to infer prevalence dynamics using 101 EBOV genomes sequenced from <b>Kailahun District</b> in Sierra Leone during the 2014-2015 EVD epidemic.</li>
    <li>Sequences from various studies, curated by <a class="cite">[Dudas et al., Nature, 2017]</a>.</li>
  </ul>
  <img data-src="WA_EVD_Dudas.jpg" style="height:350px">
</section>

<section>
  <h1>Sample distribution</h1>
  <img data-src="EbolaSampleDist.svg" style="height:500px">
  <div class="cite" style="margin-top:-25px">World Health Organization</div>

  <p style="margin-top:0">Nonlinear relationship between WHO case counts and genome samples!</p>
</section>

<section>
  <h1>Including unsequenced samples</h1>

    <ul>
    <li>Including unsequenced samples in the particle filtering algorithm is easy: just add additional observations.</li>
  </ul>

  <img data-src="seq+noseqBPF.svg" style="width:1000px">

  <ul>
    <li>Much more efficient than including unsequenced samples on tree explicitly.</li>
  </ul>
</section>

<section>
  <h1>EBOV prevalence</h1>
  <img data-src="kailahun_full.png" style="width:700px">
</section>

<section>
  <h1>EBOV reproductive number</h1>
  <img data-src="kailahun_full_Re.png" style="width:700px">
</section>

<!--
<section>
  <h1>Continuing outbreak in the DRC</h1>

  <img data-src="Ebola_Congo_spatial.png" style="height:550px">
  
</section>

<section>
  <h1>Continuing outbreak in the DRC</h1>

  <ul>
    <li>Response is primarily based on traditional epidemiological data.</li>
    <li>Increasing availability of sequencing technology may enable future
      responses to be informed by <b>real-time phylodynamics</b>.
      <ul>
        <li>Could provide up-to-the-minute estimates of where the virus has come from and to where
          it's likely to spread.</li>
      </ul>
    </li>
  </ul>

  <div>
  <img data-src="Ebola_Congo_cases.png" style="width:65%">
  <div class="cite">World Health Organization Situation Report, 3 December 2018</div>
  </div>
</section>
-->

<section>
  <div class="flipper" style="width:100%;height:800px;margin-top:-2em">
    <img data-src="AdventureMapPolytomies.png" style="height:100%;margin-top:0">
  </div>
</section>

<section>
  <h1>Recap: What are polytomies?</h1>

  <p> Polytomies are features of phylogenetic trees: internal nodes with more than two children, also known as multifurcations.</p>

  <img data-src="polytomy.svg" style="width:60%"/>

  <p style="margin-top:0em">These can represent two different things:</p>
  <ul>
    <li><b>Soft</b> polytomies merely represent an inability to resolve the order of speciation/birth events due to insufficient data.
      <li><b>Hard</b> polytomies represent a true or nearly true multifurcation of the parent species/transmission lineage.
  </ul>

  <p>We are interested in hard polytomies here.</p>
</section>

<section>
  <h1>When may polytomies occur?</h1>

  <dl>
    <dt>Gene tree context:</dt>
      <dd>Some marine species are known to have "sweepstakes reproductive success", meaning that offspring distributions have extremely high variance.</dd>
    <dt>Transmission tree context:</dt>
      <dd>Super-spreading <i>events</i> where single individuals "instantaneously" produce a large number of secondary infections.</dd>
    <dt>Species trees:</dt>
      <dd>Simultaneous speciation events.  Has been hypothesized several times in the literature.</dd>
</section>

<section>
  <h1>Polytomies in sampled trees?</h1>

  <p>Consider a population in which the distribution for the number of offspring $m$ is such that $P(m>2)>0$.</p>

  <ul>
    <li>In complete trees, polytomies will appear whenever the number
    of offspring exceeds 2.</li>
    <li>In sampled trees, polytomies will occur with vanishingly small probability.</li>
  </ul>

  <img data-src="sampled_polytomies.svg" style="width:40%"/>

  <p> $\implies$ Require offspring distribution such that number of sampled polytomies is significant.</p>
  
</section>

<section>
  <h1>Modelling polytomy generation</h1>

  <br>

  <ul>
    <li><b>Birth-death models:</b><br>
      Augment exsting birth-death models to include higher order birth events:

      $$X \overset{\lambda_m}{\longrightarrow}(m+1) X$$

      <br>

      $$X \overset{\mu}{\longrightarrow} 0$$
    </li>

    <li><b>Coalescent models:</b><br>
      The traditional coalescent excludes polytomies (almost surely).  Under the WF model, this is because higher order coalescences occur with probability $O(1/N^2)$ or lower.
    </li>
  </ul>
</section>

<section>
  <h1>The $\Lambda$-Coalescent</h1>

  <ul class="spaced">
    <li>Introduced by Sagitov (1999) and Pitman (1999).</li>
  <li>Given $n$ extant lineages, each $k$-tuple of lineages undergoes coalescence at a rate

    $$\lambda_{n,k} = \int_0^1 x^{k-2}(1-x)^{n-k}\Lambda(dx)$$

    where $\Lambda$ is a distribution on $[0,1]$.</li>
    <li>Total coalescence rate is therefore:
      $$\lambda_{\mathrm{tot}} = \sum_{k=2}^{n}\binom{n}{k}\lambda_{n,k}$$</li>

    <li>Regular coalescent corresponds to the case when $\Lambda(x)=\delta(x)$.</li>
    </ul>

</section>

<section>
  <h1>The $\Lambda$-Coalescent (ctd.)</h1>

  <ul>
    <li>Pitman (1999) derived the coalescent rates using a very general argument.</li>
    <li>Requires only that the inclusion of lineages in a merger event
    is determined by exchangeable Bernoulli trials.</li>
    <li>Together with a consistency requirement:
      $$\lambda_{n,k} = \lambda_{n+1,k} + \lambda_{n+1,k+1}$$
      and de Finetti&#39;s theorem, the rates must be those quoted on the previous slide.</li>
    <li>Equivalent to selecting $k-2$ of each of the $n-2$ remaining lineages with probability $x\sim\Lambda(x)$.</li>
    <li>Note that these "rates" are probabilities conditional on a pairwise coalescence occuring. The event rate per generation is $c_N$: the per-generation pairwise coalescence probability.</li>
  </ul>
    
</section>

<section>
  <h1>The Wright-Fisher Process</h1>

  <p>Every generation constrained to have the same size.</p>

  <img data-src="WF.svg" style="width:100%">

</section>

<section>
  <h1>The Galton-Watson Process</h1>

  <p>Every individual produces an i.i.d. number of offspring, with $\mathbb{E}[m]>1$.</p>

  <img data-src="GW.svg" style="width:100%">

  <p>Constraint: Each new generation is sampled (without replacement) from
  the intermediate generation.</p>

</section>

<section>
  <h1>The $\beta$-Coalescent</h1>

  <p>When the probability of $\geq n$ offspring decays as $1/n^{\alpha}$, the
    coalescent limit yields <a class="cite">(Schweinsberg 2003)</a>
    a $\Lambda$-coalescent with:
    $$\Lambda(x) = \frac{x^{1-\alpha}(1-x)^{\alpha-1}}{B(2-\alpha,\alpha)}$$
    where $B(q,r) = \Gamma(q)\Gamma(r)/\Gamma(q+r)$</p>

  <p>This implies that the coalescent rates become
    $$\lambda_{n,k} = \frac{B(k-\alpha,n-k+\alpha)}{B(2-\alpha,\alpha)}$$</p>
  

  <p>Taking the limit $\alpha\rightarrow 2$ yields the Kingman coalescent.</p>

    <p>Time is measured in units of $N/\sigma^2\equiv N_e$ generations, where $\sigma^2$ is the variance of the offspring distribution.</p>
</section>

<section>
  <h1>Example $\beta$-coalescent trees</h1>
  <p>$\alpha=2.0$ (regular coalescent):</p>
  <img data-src="simtree_alpha2.svg" style="width:70%">
</section>

<section>
  <h1>Example $\beta$-coalescent trees</h1>
  <p>$\alpha=1.5$</p>

  <img data-src="simtree_alpha1.5.svg" style="width:70%">
</section>

<section>
  <h1>Example $\beta$-coalescent trees</h1>
  <p>$\alpha=1.0$</p>

  <img data-src="simtree_alpha1.svg" style="width:70%">
</section>

<section>
  <h1>The $\beta$-coalescent density</h1>

  <ul>
    <li>Transforming time $dt = dt'/N_e(t')$ allows us to account for population dynamics.</li>
    <li>For a given tree, the likelihood for $\alpha$ seems close to true values.</li>
  </ul>

  <img data-src="likelihoods.png" style="width:100%">
</section>

<section>
  <h1>Inference Strategy</h1>

  <p>Usual phylogenetic posterior:</p>
  
    <blockquote style="width:100%">
      $$P(T,\mu,\theta,\alpha|A) = \frac{1}{P(A)} P(A|T,\mu)P(T|N_e,\alpha)P(\mu,N_e,\alpha)$$
    </blockquote>

    <ul>
      <li>$P(T|N_e,\alpha)$ is the $\Lambda$-coalescent tree prior:
        $$\prod_i\frac{\lambda_{n_i,k_i}}{N_e}\exp\left[-\frac{\tau_i}{N_e}\sum_{k&#39;=2}^{n_i}\binom{n_i}{k&#39;}\lambda_{n_i,k&#39;}\right]$$
      </li>
      <li>$P(\mu,N_e,\alpha)=P(\mu)P(N_e)P(\alpha)$ are the parameter priors.</li>
    </ul>

    <p> Space of trees with (potential) polytomes $\neq$ space of binary trees. </p>
    <p> Need a reversible-jump MCMC algorithm to sample time trees under this model.</p>
</section>

<section>
  <h1>Previous work</h1>

  <ul style="margin-top:20%">
    <li><a href="https://academic.oup.com/sysbio/article/54/2/241/2842885" class="cite">Lewis, Holder and Holsinger, Sys Bio, 2005</a> (unrooted trees)</li>
    <li><a href="https://doi.org/10.1016/j.tpb.2016.05.002" class="cite">Guindon, Guo and Welch, Theor Pop Biol, 2016</a> (spatial continuum model)</li>
  </ul>
</section>


<section>
  <h1>Proposal distributions:<br> Subtree Prune+Regraft</h1>

  <img data-src="proposal_SPR.svg" style="width:100%">

  <p>Tuning parameters:</p>
  <ul>
    <li>Exponential rate for root attachments.</li>
    <li>Coalescent attachment probability.</li>
  </ul>
</section>

<section>
  <h1>Proposal distributions:<br> Narrow Exchange</h1>

  <img data-src="proposal_NX.svg" style="width:100%">

</section>

<section>
  <h1>Proposal distributions:<br> Subtree Slide</h1>

  <img data-src="proposal_SS.svg" style="width:100%">

  <p>Tuning parameters:</p>
  <ul>
    <li>Rate for non-coalescent attachment..</li>
    <li>Coalescent attachment probability.</li>
  </ul>
</section>

<section>
  <h1>Proposal distributions:<br> Expand/Collapse</h1>

  <img data-src="proposal_XC.svg" style="width:100%">

  <p>Tuning parameters:</p>
  <ul>
    <li>Exponential rate for expantion above root.</li>
  </ul>
</section>

<section>
  <h1>Validation: Sampling from prior</h1>

  <img data-src="prior_sampling_heights.png" style="width:45%;padding-top:0;">
  <img data-src="prior_sampling_polytomies.png" style="width:45%;padding-top:0;">
</section>

<section>
  <h1>Validation: Inference from simulated data</h1>

  <div style="margin-top:-1em">
  <img data-src="simulated_truth.png" style="width:33%">
  <img data-src="simulated_inference.png" style="width:33%">
  </div>

</section>

<section>
  <h1>Validation: Inference from simulated data</h1>

  <img data-src="alpha_posteriors.png" style="width:100%">
</section>

<section>
  <h1>Investigation: does structure confound inference of $\alpha$?</h1>

  <p style="margin-bottom:0.5em">Simulation study:</p>

  <ul class="spaced">
    <li>Simulated trees with 60 taxa under the 3-deme structured coalescent (equal population sizes and migration rates, even leaf distribution among types).</li>
    <li>Varied migration rate from very low (highly structured trees) to very high (no discernable structure).</li>
    <li>Simulated sequences down each tree to produce alignments.</li>
    <li>Analyzed each alignment under the beta-coalescent.</li>
    </ul>
</section>

<section>
  <h1>Example structured trees</h1>

  <img data-src="structured_tree_0.1.png" style="width:45%">
  <img data-src="structured_tree_1.6.png" style="width:45%">
</section>

<section>
  <h1>Inferred beta-coalescent parameters</h1>

  <img data-src="structure_comparison.png" style="width:100%">

  <p style="margin-top:-0.5em">$\implies$ little/no confusion between structure and multifurcation?</p>
</section>

<section>
  <h1>BEAST package</h1>


  <ul class="spaced" style="margin-top:5%">
    <li>Pitchfork: <a href="http://github.com/tgvaughan/pitchfork">github.com/tgvaughan/pitchfork</a></li>
    <li>Easy BEAUti interface, integrates with other models easily.</li>
    <li>Infer $\alpha$ and (parametric) population dynamics jointly together with multifurcating trees.</li>
    <li>Help identify superspreading in transmission trees?</li>
  </ul>
</section>

<section>
  <h1>Application to Ebola</h1>

  <ul>
    <li>Again use the 2014-2015 West African Ebola virus data curated by Gytis Dudas et al. <a class="cite">(Nature, 2017)</a>.</li>
    <div style="text-align:center">
      <iframe width="500" height="300" src="https://www.youtube.com/embed/j4Ut4krp8GQ" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
    </div>
    <li>Restrict analysis to oubtreak within the Kailahun region.</li>
    <li>How do results from inference under the Beta-coalescent differ from results inferred under classic birth-death models?</li>
  </ul>
</section>

<section>
  <h1>Application to Ebola</h1>

  <img data-src="ebola_nopoly.png" style="width:70%">
  <p>Tree inferred under stochastic SIR birth-death model.</p>
</section>

<section>
  <h1>Application to Ebola</h1>

  <img data-src="ebola_poly.png" style="width:70%">
  <p>Tree inferred under constant-$N_e$ Beta-coalescent model.</p>
</section>

<section>
  <h1>Application to Ebola</h1>

  <img data-src="alpha_ucldMean_posterior_ebov.png" style="width:60%">
</section>

<section>
  <h1>Steps necessary for "serious" phylodynamic analyses</h1>

  <ul class="spaced" style="margin-top:5%">
    <li>Account for structure and population dynamics.</li>
    <li>Allow for mixture of offspring distributions:
      $$\Lambda(x) = p_0\delta(x) + (1-p_0)Beta(2-\alpha,\alpha)(x)$$</li>
    <li>Apply a spike and slab prior to $p_0$: allows for posterior probability
      on the presence of non-Kingman dynamics.</li>
    <li>These steps are complicated by the dependence of the $\beta$-coalescent timescale
      on the offspring distribution through $\alpha$.
  </ul>
</section>

<section>
  <h1>Summary</h1>
  
  <ul class="spaced">

    <li>Populations where individuals may occasionally produce a significant portion of the following generation lead to the possibility of observable polytomies in sampled trees.</li>

    <li>Bayesian phylogenetic inference under such models can be performed using a reversible-jump MCMC algorithm.</li>

    <li>Numerical results suggest that it is possible to recover parameters of the offsping distribution from genetic data.</li>

    <li>Analysis of Ebola virus data shows that this model predicts transmission trees with multiple polytomies, although interpretation requires care.</li>
  </ul>
</section>

<section>
  <h1>Work in progress: Development of birth-death model</h1>

  <p style="font-size:0.8em"> Consider the following birth/death model:
      $$X \overset{\lambda_m(x)}{\longrightarrow} (m+1)X$$
      $$X \overset{\mu}{\longrightarrow} 0$$
    where
    $$\lambda_m(x)=\lambda e^{-x}\frac{x^m}{m!}$$
    </p>

  <p style="font-size:0.8em">The probability of an individual at time $t$ in the past giving rise to $i$ lineages in the present $p_i(t)$ is:
  \begin{align}
  \frac{d}{dt}p_0(t) &= -(\lambda+\mu)p_0(t) + \mu + \lambda e^{x(p_0(t)-1)}p_0(t)\\
  \frac{d}{dt}p_1(t) &= -(\lambda+\mu)p_1(t) + \lambda(x p_0(t)+1)e^{x(p_0(t)-1)}p_1(t)
  \end{align}
  </p>

  <p style="font-size:0.8em">Sadly I can find no analytical solution to these.  Thus we need numerical integration.</p>
</section>


<section>
  <div class="flipper" style="width:100%;height:800px;margin-top:-2em">
    <img data-src="AdventureMapBacter.png" style="height:100%;margin-top:0">
  </div>
</section>


<section>
    <h1>Cartoon bacterial population genetics</h1>

    <div style="position: relative; top:0; left:0; width:1000px; height:400px">
        <img data-src="binary_fission_7.svg" class="fragment current-visible" data-fragment-index="1"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_6.svg" class="fragment" data-fragment-index="2"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_5.svg" class="fragment" data-fragment-index="3"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_4.svg" class="fragment" data-fragment-index="4"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_3.svg" class="fragment" data-fragment-index="5"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_2.svg" class="fragment" data-fragment-index="6"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
        <img data-src="binary_fission_1.svg" class="fragment" data-fragment-index="7"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none">
    </div>

    <img data-src="transduction_cropped.jpg" class="fragment" data-fragment-index="5"
                                                              style="position:absolute; top:100px; left:0px; height:250px;">

    <img data-src="conjugation.jpg" class="fragment" data-fragment-index="7"
                                                     style="position:absolute; top:100px; right:0px; height:250px;">
    <ul>
        <li class="fragment" data-fragment-index="7">Effect of events can be plasmid transfer, insertion or homologous recombination.</li>
        <li class="fragment" data-fragment-index="7">Rate of recombination varies and is subject to selection.</li>
        <li class="fragment" data-fragment-index="7">Focus on homologous recombination: only event which doesn&#39;t alter the length of the sequence.</li>
    </ul>
</section>

<section>
    <h1>The Problem for Phylogenetic Inference</h1>

    <p>For many bacteria, the ratio between the recombination rate and the mutation rate is extremely high.</p>

    <div class="figure">
        <img data-src="r2mhist.svg" style="width:700px">
        <div class="cite">Vos and Didelot, The ISME Journal (2009)</div>
    </div>
</section>

<section>
    <h1>The Coalescent with Gene Conversion</h1>

    <div style="position: relative; top:0; left:0; width:1000px; height:500px;">
        <img data-src="WrightFisherGC1.svg"
             style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC2.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC3.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC3b.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC4b.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
        <img data-src="WrightFisherGC5b.svg" class="fragment"
            style="position:absolute; top:0; left:0; width:1000px; box-shadow:none; background-color:white;">
    </div>
    <a class="cite" href="">Wiuf, 1999</a>; <a class="cite" href="">Wiuf and Hein, 2000</a>

    <!--div class="fragment" style="position:absolute; top:100px; left:0px; width:350px">
        <table class="params" style="font-size:0.8em">
            <tr><th colspan="2">Parameters</th></tr>
            <tr> <td>$N(t)g$</td> <td>Coalescence rate</td> </tr>
            <tr> <td>$\rho_s$</td> <td>Recombination rate</td> </tr>
            <tr> <td>$\delta$</td> <td>Expected tract length</td> </tr>
        </table>
    </div-->
</section>

<section class="center">
    <blockquote class="textbox">
        <div class="title">Problem</div>
        The space of possible ancestral recombination graphs is extremely
        large. Even two samples have infinitely many distinct ancestries!
    </blockquote>
</section>

<section>
    <h1>Full ARG</h1>

    <img data-src="full_arg.svg" style="box-shadow:none;width:500px">

    <br>
    &nbsp;
</section>

<section>
    <h1>Approximation: ClonalOrigin</h1>

    <img data-src="CO_arg.svg" style="box-shadow:none;width:500px">

    <br>

    <a class="cite" href="">Didelot et al., 2010 (ClonalOrigin)</a>;
    <a class="cite" href="">Ansari and Didelot, 2014 (ClonalOrigin&#39;)</a>
</section>

<section class="center">
    <blockquote class="textbox">
        <div class="title">Our Goal</div>
        To conduct Bayesian inference of the full ARG under the ClonalOrigin
        model, optionally under non-trivial population size dynamics.
    </blockquote>
</section>

<section>
  <h1>Application to simulated data</h1>
  <p style="margin:0px">True ARG:</p>
  <img data-src="true_tree.svg" style="box-shadow:none; padding:0; margin:0; width:700px">

  <p style="margin:0px">Randomly-selected ARG from MCMC:</p>
  <img data-src="sampled_tree.svg" style="box-shadow:none; padding:0; margin:0; width:700px">
</section>


<section>
    <h1>Application to <i>Escherichia coli</i></h1>

    <div>
      <div style="display:inline-block;width:40%;vertical-align:middle">
        <img data-src="E_coli.jpg" style="width:100%">
      </div>
      <div style="display:inline-block;width:49%;vertical-align:middle">
        <ul class="spaced">
          <li>Applied method to sequences from 23 <i>E. coli</i> O157/O26 human and cattle-derived isolates collected by Tessy George (<sup>m</sup>EpiLab, Massey University, New Zealand).</li>
          <li>Isolates included
            <ul>
              <li>Shiga-toxin producing strains (+STEC)</li>
              <li>non-Shiga-toxin producing strains (-STEC)</li>
            </ul>
          <li>Sequences for 53 rMLST genes<br>(21.1 kb total) from each isolate.</li>
        </ul>
      </div>
    </div>
</section>

<section>
    <h1>Application to <i>Escherichia coli</i></h1>

    <div style="position: relative; top:0; left:0; width:1000px; height:600px;">
        <img data-src="Tessy_new_summary2.svg" style="position:absolute;top:0;left:0;width:1000px;box-shadow:none;background-color:white">
        <img class="fragment" data-src="Tessy_new_summary2_compare.svg" style="position:absolute;top:0;left:0;width:1000px;box-shadow:none;background-color:white">
        <img class="fragment" data-src="gene_flow.svg" style="position:absolute;top:0;left:0;width:250px;background-color:white;padding:20px">
    </div>
</section>

<section>
  <h1>Software implementations</h1>


  <ul class="spaced">
    <li> Each algorithm and model described in this talk is implemented as free software.</li>
    <li> Specifically, each is implemented as a BEAST 2 package:
      <ol>
        <li style="margin-top:1em">EpiInf: <a href="http://github.com/tgvaughan/pitchfork">github.com/tgvaughan/EpiInf</a></li>
        <li>Bacter: <a href="http://github.com/tgvaughan/pitchfork">github.com/tgvaughan/bacter</a></li>
      </ol>
    </li>
    <li>Both packages provide a graphical interface for configuring analyses.</li>
    <li>Tutorials available on their respective websites.</li>
  </ul>

  <center>
    <img data-src="Beast2_logo.jpg">
  </center>
</section>

<section>
  <img data-src="TTB_logo.png" style="width:70%">

  <ul>
  <li>Members of the cEvo group have organized a global series of workshops providing instruction on conducting Bayesian phylodynamic analyses using BEAST 2.</li>
  <li>Workshops have been held in
    <ul class="">
      <li>Switzerland (twice)</li>
      <li>London, UK</li>
      <li>Auckland, New Zealand</li>
      <li>Sydney, Australia</li>
      <li>Squamish, Canada</li>
      </ul>
  <li>A workshop in Switzerland will be held next June.</li>
  <li>We also maintain a growing collection of BEAST 2 tutorials and instructional material at <a href="https://taming-the-beast.org">https://taming.the.beast.org</a>.</li>
  </ul>

</section>

<section>
  <h1>Summary</h1>

  <ul class="spaced" style="margin-top:10%">
    <li>Phylogenetic inference provides much more than a means of inferring phylogenetic trees.</li>
    <li>Broadly, these methods provide a means of connecting observed genetic and other forms of data to a large variety of features of the population from which the data is sampled.</li>

    <li>Bayesian phylodynamics is the key to many adventures!</li>
  </ul>
</section>

<section>
  <h1>Acknowledgements</h1>

  <dl>
    <li><dt>Epidemic dynamics inference:</dt>
      <dd>Tanja Stadler (ETHZ), Gabriel Leventhal (MIT), David Rasmussen (NCSU),<br>
        David Welch and Alexei Drummond (Univ. Auckland, NZ)</dd></li>
    <li><dt>Polytomies:</dt>
      <dd>Oliver Pybus (Oxford), Patrick Hoscheit (INRA)</dd></li>
    <li><dt>Bacterial recombination graph inference:</dt>
      <dd>Nigel French, Tessy George and Patrick Biggs (Massey Univ., NZ)<br>
        David Welch and Alexei Drummond (Univ. Auckland, NZ)</dd></li>
    <li>Prof. Tanja Stadler&#39;s Computational Evolution (cEvo) group!</li>
  </dl>

  <center>
    <img data-src="group_jump_photo.jpg" style="height:9em">
  </center>
</section>


<!-- Outline:

* Overview slide

---

* What is a phylogenetic tree?

* What is phylodynamics?

* Big slide connecting all the dots

* Bayesian phylodynamics

---

* Modeling epidemics using birth-death branching processes

* Challenge of computing phylodynamic likelihood

* Simulation-based approach (particle filtering)

* Application to Ebola data

* Software

---

* Bacterial evolution - rampant recombination

* Phylogenetic recombination networks

* Simplified networks

* Application to E coli pathogenicity

* Software

---

* Polytomies in phylogenetic trees

* What can cause polytomies in transmission trees?

* Algorithm for inferring polytomies

* Application to Simulated data

* Application to Ebola data

* Software

---

* Summary and Acknowledgements
-->
